File: CustomView/app/src/main/java/sample/github/nisrulz/customview/CustomView.java
package sample.github.nisrulz.customview;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.view.View;

public class CustomView extends View {

  Paint paint = new Paint();

  public CustomView(Context context) {
    super(context);

    paint.setColor(Color.BLACK);
    paint.setTextSize(30);
  }

  @Override protected void onDraw(Canvas canvas) {
    super.onDraw(canvas);

    setBackgroundColor(Color.BLUE);
    canvas.drawText("Custom Text", 100, 100, paint);

  }


}

File: CustomView/app/src/main/java/github/nisrulz/example/customview/CustomView.kt
package github.nisrulz.example.customview

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

class CustomView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : View(context, attrs, defStyleAttr) {

    private var paint = Paint()
    private var _txt = "Custom Text"

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        setBackgroundColor(Color.BLUE)
        canvas.drawText(_txt, 100f, 250f, paint)
    }

    fun setText(txt: String) {
        _txt = txt
        invalidate()
    }

    init {
        paint.apply {
            color = Color.WHITE
            textSize = 150f
        }
    }
}